miden-lib/asm/miden/note.masm

```diff
@@ -143,4 +143,171 @@ export.get_serial_number
 
     syscall.get_note_serial_number
     # => [SERIAL_NUMBER]
-end
\ No newline at end of file
+end
+
+#! Computes hash of note inputs starting at the specified memory address.
+#!
+#! This procedure divides the hashing process into two parts: hashing pairs of words using 
+#! `absorb_double_words_from_memory` procedure and hashing the remaining values using the `hperm`
+#! instruction. 
+#!
+#! Inputs:  [ptr, num_elements]
+#! Outputs: [HASH]
+#! Cycles: 
+#! - If number of elements divides by 8: 56 cycles + 3 * words
+#! - Else: 189 cycles + 3 * words
+#!
+#! Panics if num_elements is equals 0 or greater than 128.
+export.compute_inputs_hash
+    # check that number of inputs is less than 128
+    dup.1 push.128 u32assert2 u32lte assert
+
+    # move number of inputs to the top of the stack 
+    swap
+    # => [num_elements, ptr]
+
+    # check that number of inputs greater than 0
+    dup eq.0 assertz
+    # => [num_elements, ptr]
```

Comment:
For hashing inputs, 0 inputs should hash to `[0, 0, 0, 0]` - so, we need to update the logic and also the procedure description.